src/compiler/compile/render_dom/wrappers/EachBlock.ts

```diff
@@ -445,13 +445,18 @@ export default class EachBlockWrapper extends Wrapper {
 					}
 				`
 				: deindent`
-					${iterations}[#i] = ${create_each_block}(child_ctx);
-					${iterations}[#i].c();
-					${has_transitions && `@transition_in(${this.vars.iterations}[#i], 1);`}
-					${iterations}[#i].m(${update_mount_node}, ${anchor});
+					if (${iterations}[#i]) {
+						${has_transitions && `@transition_in(${this.vars.iterations}[#i], 1);`}
```

Comment:
This looks like it's going to create an avoidable empty `if` branch if there are not transitions.

Reply:
ah, yeah. fixed (albeit slightly clumsily — I keep coming back to the conclusion that we probably need a slightly smarter approach to code generation)